### DISCUSSION TITLE
How should we handle branches within projects in Nitric Deploy

### DISCUSSION THREAD

**Discussion by raksiv** [nitrictech/nitric#325] [Q&A]
Currently Nitric Deploy allows users to configure a single production branch, when a user commits to the branch, the stack (app code and infrastructure) is automatically updated. 

Before we make any more changes to how we handle branches, let's have a community discussion to learn what users need out of this feature.

- Should there be only 1 production branch? If not, what are the use cases which require more than one?
- Should this production branch be a mandatory setting?
- Should multiple product branches == multiple projects (one production per project)

**Comment by tjholm** [comment]
@raksiv a single production branch makes sense in the case where nitric deploy would support custom domains or vanity urls for created apis.

So when a custom domain is configured, it would point to that deployment by default, and other deployments would probably include their stack name as part of the custom domain.

e.g.
Projects Custom domain  = my-api.example.com
Prod -> my-api.example.com
Dev -> my-api-dev.example.com
